amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/model/domain/AnnotationMapping.scala

```diff
@@ -1,20 +1,31 @@
 package amf.plugins.document.vocabularies.model.domain
 
 import amf.core.model.StrField
+import amf.core.model.domain.{DomainElement, Linkable}
 import amf.core.parser.{Annotations, Fields}
 import amf.plugins.document.vocabularies.metamodel.domain.AnnotationMappingModel
 import amf.plugins.document.vocabularies.metamodel.domain.AnnotationMappingModel._
 import org.yaml.model.YMap
 
 class AnnotationMapping(override val fields: Fields, override val annotations: Annotations)
-    extends PropertyLikeMapping[AnnotationMappingModel.type] {
+    extends PropertyLikeMapping[AnnotationMappingModel.type]
+    with NodeMappable[AnnotationMappingModel.type] {
+
+  override def name: StrField                    = super[NodeMappable].name
```

Comment:
Why overriding these methods?

Reply:
I had to add NodeMappable to the class hierarchy. Both PropertyLikeMapping and NodeMappable implement the `name` and `withName` methods so I had to override them in the extending class.